tests/acceptance/features/bootstrap/GraphContext.php

```diff
@@ -2317,4 +2317,121 @@ public function userGetsThePersonalDriveInformationOfUserUsingGraphApi(string $b
 			)
 		);
 	}
+
+	/**
+	 * @When /^the user "([^"]*)" unassigns the role of user "([^"]*)" using the Graph API$/
+	 * @When /^the user "([^"]*)" tries to unassign the role of user "([^"]*)" using the Graph API$/
+	 *
+	 * @param string $user
+	 * @param string $ofUser
+	 *
+	 * @return void
+	 *
+	 * @throws GuzzleException
+	 * @throws Exception
+	 */
+	public function theUserUnassignsTheRoleOfUserUsingTheGraphApi(string $user, string $ofUser): void {
+		$userId = $this->featureContext->getAttributeOfCreatedUser($ofUser, 'id') ?? $ofUser;
+		$credentials = $this->getAdminOrUserCredentials($user);
+
+		$response = GraphHelper::getAssignedRole(
+			$this->featureContext->getBaseUrl(),
+			$this->featureContext->getStepLineRef(),
+			$this->featureContext->getAdminUsername(),
+			$this->featureContext->getAdminPassword(),
+			$userId
+		);
+		$appRoleAssignmentId = $this->featureContext->getJsonDecodedResponse($response)["value"][0]["id"];
+
+		$this->featureContext->setResponse(
+			GraphHelper::unassignRole(
+				$this->featureContext->getBaseUrl(),
+				$this->featureContext->getStepLineRef(),
+				$credentials['username'],
+				$credentials['password'],
+				$appRoleAssignmentId,
+				$userId
+			)
+		);
+	}
+
+	/**
+	 * @Then user :user should have the role :role assigned
+	 *
+	 * @param string $user
+	 * @param string $role
+	 *
+	 * @return void
+	 * @throws GuzzleException
+	 * @throws Exception
+	 */
+	public function userShouldHaveTheRoleAssigned(string $user, string $role): void {
+		$userId = $this->featureContext->getAttributeOfCreatedUser($user, 'id') ?? $user;
+		$response = GraphHelper::getAssignedRole(
+			$this->featureContext->getBaseUrl(),
+			$this->featureContext->getStepLineRef(),
+			$this->featureContext->getAdminUserName(),
+			$this->featureContext->getAdminPassword(),
+			$userId
+		);
+
+		$jsonDecodedResponse = $this->featureContext->getJsonDecodedResponse($response)['value'][0];
```

Comment:
maybe we could use store user-role id in
```
private $appEntity = [];
```
to avoid making API request just to get user-role id

Reply:
@grgprarup any specific reason for not implementing suggestion

Reply:
I think here, the current role-Id of a user (`$user`) is fetched and asserted that the id matches the expected role (`$role`) id.

Reply:
AFAIK The purpose of appEntity is to store application roles, which is not used to store each user's role.

Reply:
Yeah,
Stores Ids for different of types of roles